docs/docs/03-language-reference.md

```diff
@@ -572,20 +572,27 @@ The following is a list of supported indexable types:
 
 [`▲ top`][top]
 
-### 1.2 Utility Functions
-
-| Name         | Extra information                                     |
-| ------------ | ----------------------------------------------------- |
-| `log`        | logs str                                              |
-| `assert`     | checks a condition and _throws_ if evaluated to false |
-| `unsafeCast` | cast a value into a different type                    |
-| `nodeof`     | obtain the [tree node](./02-concepts/02-application-tree.md) of a preflight object |
-| `lift`       | explicitly qualify a [lift](./02-concepts/01-preflight-and-inflight.md#explicit-lift-qualification) of a preflight object |
+### 1.2 Intrinsic Functions
+
+Intrinsic functions are a special call-like expressions built into the Wing compiler with
+the following properties (given an example intrinsic `@x()`):
+- It is not possble to write `let variable = @x;` or otherwise use `@x` as an expression on its own
+- `x` is not automatically a symbol that can be referenced
+- The arguments/return types must be representable Wing types, but can be more dynamic than user-defined functions
+  - For example, the return type may change between inflight and preflight
+
+| Name            | Extra information                                                                                                         |
+| --------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| `@log()`        | logs str                                                                                                                  |
```

Comment:
It should be okay to allow the previous "global functions" and these new intrinsics to coexist during deprecation, sure. I figured the spec could focus only on the future state.

If you're proposing that `log()` should always exist, then that's worth discussing. I prefer killing all the magical globals, and pragmatically the extra `@` isn't sufficient friction to be concerned.
99% percent of the time I expect the keystroke count to remain the same. With `log()`, my keystrokes will often be `<l> <o> <g> <enter>` because there can be conflicting symbols. With `@log()`, I expect to only type `<@> <l> <o> <enter>` because there are less options after `@`. As a pragmatic choice we can even make `@log` come before `@lift` in the completions, then you would only type `<@> <l> <enter>`

Reply:
I am okay with that, but I think @staycoolcall911 had some concerns.

Definitely the soft-deprecation will allow us to experiment with this and see how it feels.